src/fractal_faim_hcs/MetaSeriesUtils_dask.py

```diff
@@ -0,0 +1,302 @@
+    # -> maybe implement checks
+
+    planes = sorted(well_files["z"].unique(), key=int)
+
+    fields = sorted(
+        well_files["field"].unique(),
+        key=lambda s: int(re.findall(r"(\d+)", s)[0])
+    )
+
+    # Create an empty np array to store the filenames in the correct structure
+    fn_dtype = f"<U{max([len(fn) for fn in well_files['path']])}"
+    fns_np = np.zeros(
+        (len(fields), len(channels), len(planes)),
+        dtype=fn_dtype,
+    )
+
+    # Store fns in correct position
+    for s, field in enumerate(fields):
+        field_files = well_files[well_files["field"] == field]
+        for c, channel in enumerate(channels):
+            channel_files = field_files[field_files["channel"] == channel]
+            for z, plane in enumerate(planes):
+                plane_files = channel_files[channel_files["z"] == plane]
+                if len(plane_files) == 1:
+                    fns_np[s, c, z] = list(plane_files["path"])[0]
+                elif len(plane_files) > 1:
+                    raise RuntimeError("Multiple files found for one FCZ")
+
+    # distinguish between channels with
+    #   - all z-planes
+    #   - only one z-plane (stored in first plane)
+    #   - no z-planes (only projections)
+    channels_full_planes = []
+    channels_one_plane = []
+    channels_empty = []
+    for c in range(fns_np.shape[1]):
+        if '' not in fns_np[:,c,:]:
+            channels_full_planes.append(c)
+        elif '' not in fns_np[:,c,0]:
+            channels_one_plane.append(c)
+        else:
+            channels_empty.append(c)
+
+
+    # LOAD STAGE POSITIONS:
+    # TODO: maybe only read z-positions
+    def _da_read_stage_positions(x):
+        fn = x.flatten()[0]
+        if fn=='':
+            raise RuntimeError('File not found. This channel probably either only has projections or only one plane.')
+        ms_metadata = load_metaseries_tiff_metadata(fn)
+        x_pos = ms_metadata["stage-position-x"]
+        y_pos = ms_metadata["stage-position-y"]
+        z_pos = ms_metadata["z-position"]
+        output = np.array([x_pos, y_pos, z_pos], dtype="float64")
+        newshape = x.shape + output.shape
+        return np.reshape(output, newshape)
+
+    fns_da = da.from_array(
+        fns_np,
+        chunks=((1,)*len(fns_np.shape))
+    )
+
+    stage_positions_da = da.map_blocks(
+        _da_read_stage_positions,
+        fns_da,
+        chunks=da.core.normalize_chunks(
+            (1,) * len(fns_da.shape) + (3,), (*fns_da.shape, 3)
+        ),
+        new_axis=len(fns_da.shape),
+        meta=np.asanyarray([]).astype("float64"),
+    )
+
+    z_positions_full_planes = da.mean(
+        stage_positions_da[:,channels_full_planes,:,2],
+        axis=0
+    ).compute()
+    z_positions_one_plane = da.mean(
+        stage_positions_da[:,channels_one_plane,0,2],
+        axis=0
+    ).compute()
+
+    z_sampling = compute_z_sampling(z_positions_full_planes)
```

Comment:
A bit confusing that we're calculating z_sampling here, but then do another approach to place the planes and only use the sampling further down. Can we simplify this?